File: backends/ruby-rails/app/domain/notification/entity.rb
# frozen_string_literal: true
#
# 通知エンティティを表すドメインオブジェクトモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Notification
    # 通知のエンティティです。
    Entity = Struct.new(
      :id, :staff_id, :message_type, :title, :message, :url, :read,
      :created_at, :created_by, :updated_at, :updated_by,
      :deleted_at, :deleted_by, :version,
      keyword_init: true
    )
  end
end

File: backends/rust-axum/src/domain/notification/entity.rs
//! 通知ドメイン エンティティモジュール。
//!
//! # Author
//! Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

use chrono::{DateTime, Utc};

/// 通知のドメインエンティティ。
pub struct Notification {
    pub id:           u64,
    pub staff_id:     u32,
    pub message_type: i32,
    pub title:        String,
    pub message:      String,
    pub url:          Option<String>,
    pub read:         bool,
    pub created_at: DateTime<Utc>,
    pub created_by:   Option<u32>,
    pub updated_at: DateTime<Utc>,
    pub updated_by:   Option<u32>,
    pub deleted_at: Option<DateTime<Utc>>,
    pub deleted_by:   Option<u32>,
    pub version:      i32,
}
